From 677e9677e6f03ef82e6bdec455d82b8baa537c02 Mon Sep 17 00:00:00 2001
From: Daniel Zhang <danielzhang130@gmail.com>
Date: Wed, 8 Feb 2023 17:38:37 -0500
Subject: [PATCH 1/2] Update device config

- enable precise location prompt
- enable 7 day history in privacy dashboard

Change-Id: I3e8487edc568209f1a674fcbbe817344d36f8178
Signed-off-by: Daniel Zhang <danielzhang130@gmail.com>
---
 .../permissioncontroller/permission/utils/KotlinUtils.kt   | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/PermissionController/src/com/android/permissioncontroller/permission/utils/KotlinUtils.kt b/PermissionController/src/com/android/permissioncontroller/permission/utils/KotlinUtils.kt
index 2aad015..de6f116 100644
--- a/PermissionController/src/com/android/permissioncontroller/permission/utils/KotlinUtils.kt
+++ b/PermissionController/src/com/android/permissioncontroller/permission/utils/KotlinUtils.kt
@@ -214,12 +214,7 @@ object KotlinUtils {
      */
     @ChecksSdkIntAtLeast(Build.VERSION_CODES.S)
     fun is7DayToggleEnabled(): Boolean {
-        return SdkLevel.isAtLeastS() &&
-            DeviceConfig.getBoolean(
-                DeviceConfig.NAMESPACE_PRIVACY,
-                PRIVACY_DASHBOARD_7_DAY_TOGGLE,
-                false
-            )
+        return true
     }
 
     /**
-- 
2.34.1


From 3bff226c2f0e506ac01a7d742b8b712bec7bb6fa Mon Sep 17 00:00:00 2001
From: Daniel Zhang <danielzhang130@gmail.com>
Date: Fri, 29 Dec 2023 22:28:13 -0500
Subject: [PATCH 2/2] Make FAKE_PACKAGE_SIGNATURE runtime permission

Change-Id: If9ded0a8d6b8dd7044417e9319c413e4927fb23a
Signed-off-by: Daniel Zhang <danielzhang130@gmail.com>
---
 .../permissioncontroller/permission/utils/PermissionMapping.kt   | 1 +
 1 file changed, 1 insertion(+)

diff --git a/PermissionController/src/com/android/permissioncontroller/permission/utils/PermissionMapping.kt b/PermissionController/src/com/android/permissioncontroller/permission/utils/PermissionMapping.kt
index 840a033..b51e1a9 100644
--- a/PermissionController/src/com/android/permissioncontroller/permission/utils/PermissionMapping.kt
+++ b/PermissionController/src/com/android/permissioncontroller/permission/utils/PermissionMapping.kt
@@ -175,6 +175,7 @@ object PermissionMapping {
         }
 
         PLATFORM_PERMISSIONS[Manifest.permission.BODY_SENSORS] = Manifest.permission_group.SENSORS
+        PLATFORM_PERMISSIONS[Manifest.permission.FAKE_PACKAGE_SIGNATURE] = Manifest.permission_group.FAKE_PACKAGE
 
         if (SdkLevel.isAtLeastT()) {
             PLATFORM_PERMISSIONS[Manifest.permission.POST_NOTIFICATIONS] =
-- 
2.34.1

